spatialos-sdk/src/worker/schema/component_update.rs

```diff
@@ -0,0 +1,185 @@
+use crate::worker::{
+    component::{Component, ComponentId, Update},
+    schema::{owned::*, ArrayField, FieldId, Object, SchemaField, SchemaObjectType},
+};
+use spatialos_sdk_sys::worker::*;
+use std::marker::PhantomData;
+
+#[derive(Debug)]
+pub struct ComponentUpdate(PhantomData<*mut Schema_ComponentUpdate>);
+
+impl ComponentUpdate {
+    pub fn new<U: Update>(update: &U) -> Owned<Self> {
+        // Create the underlying `Schema_ComponentUpdate` and wrap it in a smart pointer.
+        let mut result: Owned<Self> =
+            unsafe { Owned::new(Schema_CreateComponentUpdate(U::Component::ID)) };
+
+        // Populate the update object from the update data.
+        let component_update = &mut *result;
+        update.into_update(component_update);
+
+        result
+    }
+
+    pub(crate) unsafe fn from_raw<'a>(raw: *mut Schema_ComponentUpdate) -> &'a Self {
+        &*(raw as *mut _)
+    }
+
+    pub fn component_id(&self) -> ComponentId {
+        unsafe { Schema_GetComponentUpdateComponentId(self.as_ptr()) }
+    }
+
+    /// Deserializes the component update into the specified update type.
+    ///
+    /// Returns `None` if `U` is not the correct update type.
+    pub fn deserialize<U: Update>(&self) -> Option<U> {
+        if self.component_id() == U::Component::ID {
+            Some(U::from_update(self))
+        } else {
+            None
+        }
+    }
+
+    pub fn field<T: SchemaField>(&self, field: FieldId) -> Option<T::RustType> {
+        T::get_update(self, field)
+    }
+
+    pub fn add_field<T: SchemaField>(&mut self, field: FieldId, value: &T::RustType) {
+        T::add_update(self, field, value);
+    }
+
+    pub fn field_array<T: ArrayField>(&self, field: FieldId) -> Option<Vec<T::RustType>> {
+        if T::has_update(self, field) {
+            Some(self.fields().field_array::<T>(field))
+        } else {
+            None
+        }
+    }
+
+    pub fn add_field_array<T: ArrayField>(&mut self, field: FieldId, value: &[T::RustType]) {
+        self.fields_mut().add_field_array::<T>(field, value)
+    }
+
+    pub fn add_cleared(&mut self, field: FieldId) {
+        unsafe {
+            Schema_AddComponentUpdateClearedField(self.as_ptr(), field);
+        }
+    }
+
+    pub fn field_cleared(&self, field: FieldId) -> bool {
+        // TODO: Implement this using `Schema_IsComponentUpdateFieldCleared` once we update
```

Comment:
If you rebase this branch once I merge the 14.0 PR, you can use the `Schema_IsComponentUpdateFieldCleared` API.